apps/src/pythonlab/pyodideRunner.ts

```diff
@@ -54,7 +53,19 @@ export async function runPythonCode(mainFile: string, source: MultiFileSource) {
   }
 }
 
-export async function runAllTests(source: MultiFileSource) {
-  // To run all tests in the project, we look for files that follow the regex 'test*.py'
-  await runPythonCode(getTestRunnerScript('test*.py'), source);
+export async function runAllTests(
+  source: MultiFileSource,
+  dispatch: Dispatch<AnyAction>
+) {
+  // If the project has a validation file, we just run those tests.
+  const validationFile = Object.values(source.files).filter(f => f.validation);
+  if (validationFile.length > 0) {
```

Comment:
same as above, this could be a `find`

```
const validationFile = Object.values(source.files).filter(f => f.validation);
```

It'll automatically stop on the first one, so the behavior is the same. It slightly masks the concept that there _might_ be more than one validation file, so it still might be useful to call that out in a comment.

Reply:
will update!